be/src/runtime/task_group/task_group.cpp

```diff
@@ -266,8 +273,167 @@ Status TaskGroupInfo::parse_topic_info(const TWorkloadGroupInfo& workload_group_
+                std::make_unique<vectorized::SimplifiedScanScheduler>("RScan_" + tg_name,
+                                                                      cg_cpu_ctl_ptr);
+        Status ret = remote_scan_scheduler->start();
+        if (ret.ok()) {
+            _remote_scan_task_sched = std::move(remote_scan_scheduler);
+        } else {
+            LOG(INFO) << "[upsert wg thread pool] remote scan scheduler start failed, gid="
+                      << tg_id;
+        }
+    }
+    if (max_remote_scan_thread_num > 0 && _remote_scan_task_sched) {
+        _remote_scan_task_sched->reset_max_thread_num(max_remote_scan_thread_num);
+    }
+    if (min_remote_scan_thread_num > 0 && _remote_scan_task_sched) {
+        _remote_scan_task_sched->reset_min_thread_num(min_remote_scan_thread_num);
+    }
+
+    if (_non_pipe_thread_pool == nullptr) {
+        std::unique_ptr<ThreadPool> thread_pool = nullptr;
+        auto ret = ThreadPoolBuilder("nonPip_" + tg_name)
+                           .set_min_threads(1)
+                           .set_max_threads(config::fragment_pool_thread_num_max)
+                           .set_max_queue_size(config::fragment_pool_queue_size)
+                           .set_cgroup_cpu_ctl(cg_cpu_ctl_ptr)
+                           .build(&thread_pool);
+        if (!ret.ok()) {
+            LOG(INFO) << "[upsert wg thread pool] create non-pipline thread pool failed, gid="
+                      << tg_id;
+        } else {
+            _non_pipe_thread_pool = std::move(thread_pool);
+        }
+    }
+
+    // step 6: update cgroup cpu if needed
+    if (_cgroup_cpu_ctl) {
+        if (enable_cpu_hard_limit) {
+            if (cpu_hard_limit > 0) {
+                _cgroup_cpu_ctl->update_cpu_hard_limit(cpu_hard_limit);
+                _cgroup_cpu_ctl->update_cpu_soft_limit(CPU_SOFT_LIMIT_DEFAULT_VALUE);
+            } else {
+                LOG(INFO) << "[upsert wg thread pool] enable cpu hard limit but value is illegal: "
+                          << cpu_hard_limit << ", gid=" << tg_id;
+            }
+        } else {
+            if (config::enable_cgroup_cpu_soft_limit) {
+                _cgroup_cpu_ctl->update_cpu_soft_limit(cpu_shares);
+                _cgroup_cpu_ctl->update_cpu_hard_limit(
+                        CPU_HARD_LIMIT_DEFAULT_VALUE); // disable cpu hard limit
+            }
+        }
+        _cgroup_cpu_ctl->get_cgroup_cpu_info(&(tg_info->cgroup_cpu_shares),
+                                             &(tg_info->cgroup_cpu_hard_limit));
+    }
+}
+
+void TaskGroup::get_query_scheduler(doris::pipeline::TaskScheduler** exec_sched,
+                                    vectorized::SimplifiedScanScheduler** scan_sched,
+                                    ThreadPool** non_pipe_thread_pool,
+                                    vectorized::SimplifiedScanScheduler** remote_scan_sched) {
+    std::shared_lock<std::shared_mutex> rlock(_task_sched_lock);
+    *exec_sched = _task_sched.get();
```

Comment:
Should check if the task sched or scan task sched is valid. If it is not valid then should return the global sched thread pool.

Reply:
the logic ```return global sched```  is in get_pipe_exec_scheduler/get_non_pipe_exec_thread_pool